### ISSUE TITLE
Feature: Availability of arguments

### ISSUE THREAD

**Issue by jeffasinger** [paolo-losi/stormed-amqp#15]
Several different methods in AMQP have a table of arguments. For example, exchange.declare, and queue.declare.

In each of these cases, arguments is sent as an empty dict to FrameHandler.sendMethod. It would be helpful if these could be sent on if arguments was set.

However, care needs to be taken with serialization, because the type being sent to the server would often matter.

**Comment by sjlongland** [comment]
I might add to this, some of these methods also have parameters that can be passed in, e.g. exchange_declare has a `passive` flag.

http://www.rabbitmq.com/amqp-0-9-1-quickref.html#exchange.declare

I've just discovered this morning that stormed-amqp does not support this in release 0.2.

**Comment by paolo-losi** [comment]
@jeffasinger you're right table are not fully supported right now in stormed.
They are not exposed as arguments as you notices and if exposed they would be currently
limited to string type values. Have you got a specific use case in mind. If string values tables are
enough it would be straight-forward to implement it

@sjlongland in the case of passive flag I've opted to keep the api "small". You can reach exactly the
same result for passive=True when you don't provide a callback. You certainly get more traffic (exchange.declare-ok) but that shouldn't be a problem if arguably the configuration traffic is less that the "real" traffic. What do you think?

**Comment by jeffasinger** [comment]
@paolo-losi We've actually dumped using AMQP in that part of our app altogether. I ended up forking it, but never contributing back, as the forked version didn't work 100% for all arguments, but so long as the argument type was string, it just happened to work.

The argument that I was interested in was x-message-ttl, which is specific to the RabbitMQ implementation.

**Comment by sjlongland** [comment]
Yeah, not providing a callback is one option, but I need to know that the declare succeeded so that I can proceed with a queue_bind later.

**Comment by paolo-losi** [comment]
@sjlongland if you get an error it is going to be trapped with Channel.on_error whether passive is True or not. So there is really no need to provide the passive argument.

**Comment by paolo-losi** [comment]
@jeffasinger thanks for the use case. x-message-ttl is indeed useful. I'll need to extend table support to non string types...
